### ISSUE TITLE
failed installing in AKS

### ISSUE THREAD

**Issue by ctmillerlin** [microsoft/CloudAdoptionFramework#82]
We are installing the Azure naming tool in AKS. But we failed when inputting admin password. Could you help check this?
<img width="1569" alt="image" src="https://user-images.githubusercontent.com/59555852/183585706-fc770337-1825-4561-ba6d-81508c8c7309.png">

by checking trace log
`fail: Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost[111]
      Unhandled exception in circuit 'ZBqkfabWaFBr11MunFgu4v9w0VvkMOvJzzgu99inNMI'.
      System.IO.InvalidDataException: Failed to load configuration from file '/app/settings/appsettings.json'.
       ---> System.FormatException: Could not parse the JSON file.
       ---> System.Text.Json.JsonReaderException: 'c' is invalid after a single JSON value. Expected end of data. LineNumber: 5 | BytePositionInLine: 1.
         at System.Text.Json.ThrowHelper.ThrowJsonReaderException(Utf8JsonReader& json, ExceptionResource resource, Byte nextByte, ReadOnlySpan`1 bytes)
         at System.Text.Json.Utf8JsonReader.ConsumeNextTokenUntilAfterAllCommentsAreSkipped(Byte marker)
         at System.Text.Json.Utf8JsonReader.ConsumeNextToken(Byte marker)
         at System.Text.Json.Utf8JsonReader.ReadSingleSegment()
         at System.Text.Json.Utf8JsonReader.Read()
         at System.Text.Json.JsonDocument.Parse(ReadOnlySpan`1 utf8JsonSpan, JsonReaderOptions readerOptions, MetadataDb& database, StackRowStack& stack)
         at System.Text.Json.JsonDocument.Parse(ReadOnlyMemory`1 utf8Json, JsonReaderOptions readerOptions, Byte[] extraRentedArrayPoolBytes, PooledByteBufferWriter extraPooledByteBufferWriter)
         at System.Text.Json.JsonDocument.Parse(ReadOnlyMemory`1 json, JsonDocumentOptions options)
         at System.Text.Json.JsonDocument.Parse(String json, JsonDocumentOptions options)
         at Microsoft.Extensions.Configuration.Json.JsonConfigurationFileParser.ParseStream(Stream input)
         at Microsoft.Extensions.Configuration.Json.JsonConfigurationProvider.Load(Stream stream)
         --- End of inner exception stack trace ---
         at Microsoft.Extensions.Configuration.Json.JsonConfigurationProvider.Load(Stream stream)
         at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load(Boolean reload)
         --- End of inner exception stack trace ---
         at Microsoft.Extensions.Configuration.FileConfigurationProvider.HandleException(ExceptionDispatchInfo info)
         at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load(Boolean reload)
         at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load()
         at Microsoft.Extensions.Configuration.ConfigurationRoot..ctor(IList`1 providers)
         at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build()
         at AzureNamingTool.Helpers.GeneralHelper.GetConfigurationData() in /src/Helpers/GeneralHelper.cs:line 17
         at AzureNamingTool.Shared.PasswordModal.ModalSave() in /src/Shared/PasswordModal.razor:line 61
         at Microsoft.AspNetCore.Components.ComponentBase.CallStateHasChangedOnAsyncCompletion(Task task)
         at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)
`

by checking appsettings.json, don't know why there is a `ck"` in it.
<img width="498" alt="image" src="https://user-images.githubusercontent.com/59555852/183585964-62f3fcdb-f9d2-464b-9459-2b6a2dab4b51.png">

**Comment by BryanSoltis** [comment]
Hi @ctmillerlin,

Thank you for your feedback. I just went through a deployment of the tool to AKS, following this guide [Tutorial](https://docs.microsoft.com/en-us/azure/aks/tutorial-kubernetes-prepare-app).  I created my own Azure Container Registry and pushed my image to the registry.

When I deployed it, it seemed to work correctly with no errors. In reviewing your error code, it appears as if your appsettings.json does indeed have invalid characters. I'm not sure how that happened but let me explain what happens in the code to hopefully explain how to resolve it. 

- The code repo contains a **/repository** folder that contains the initial configuration files for the site, including a "vanilla" appsettings.json file. 
- When the site is first launched, all the files in the **/repository** folder are copied to the **/settings** folder. 
- The site will generate a random API Key GUID and save it to the **/settings/appsettings.json** file.
- The site will prompt to set a password. That password is also saved to the **/settings/appsettings.json** file.
- Any subsequent updates are saved to the **/settings** version of the settings and configuration files

In the project notes, it's stated that the **/settings** folder must be mapped to a persistent storage location, so that the settings will be retained when the container is deleted/created. 

I'm still not sure how your appsettings.json file got other values (the "ck" text), but you could try one of the following to resolve the issue:

Repeat the process
- Pull the latest source code
- Create and push a new image to your container registry
- Create a new AKS cluster
- Attempt to deploy the site again

Update the appsettings.json file
- In your /settings directory, delete the **appsettings.json** file
- Access the site and confirm the **/respository/appsetting.json** file is copied over
- Attempt to access the site again

Hope this helps! 

BryanSoltis

**Comment by ctmillerlin** [comment]
@BryanSoltis Thanks for your reply. Let me try again

**Comment by ctmillerlin** [comment]
@BryanSoltis fixed by removing the appsettings.json file
